### PR TITLE
refactor: remove 'partial' ops and replace with expressions

### PR DESCRIPTION
**Pull request by TrevorBergeron**
Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [ ] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/python-bigquery-dataframes/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [ ] Ensure the tests and linter pass
- [ ] Code coverage does not decrease (if any source code was changed)
- [ ] Appropriate docs were updated (if necessary)

Fixes #<issue_number_goes_here> 🦕

### CODE REVIEW

File: bigframes/core/indexers.py [hunk 1]
```diff
@@ -63,17 +64,14 @@ def __setitem__(self, key, value) -> None:
...
-            index_column, ops.partial_right(ops.eq_op, key)
-        )
-        block, result_id = block.apply_binary_op(
-            insert_cond,
-            self._series._value_column,
-            ops.partial_arg1(ops.where_op, value),
-        )
-        block = block.copy_values(result_id, value_column).drop_columns(
-            [insert_cond, result_id]
+        blcok, result_id = block.project_expr(
```
Comment (chelsea-lin) [inline bigframes/core/indexers.py:67]: `blcok` typo?

**Review by chelsea-lin — COMMENTED** [review]
LGTM overall, except one mild comment.

File: bigframes/core/indexers.py (hunk 1 above)
Comment (TrevorBergeron) [reply bigframes/core/indexers.py]: good catch. fixed this and a few other bugs

**Review by chelsea-lin — APPROVED** [review]
LTGM!

**Pull request merged**